src/ErrorOrExtensions.cs

```diff
@@ -0,0 +1,19 @@
+namespace ErrorOr;
+
+public static class ErrorOrExtensions
+{
+    /// <summary>
+    /// If the state of <paramref name="errorOr"/> is a value, the provided function <paramref name="onValue"/> is executed asynchronously and its result is returned.
+    /// </summary>
+    /// <typeparam name="TResult">The type of the result.</typeparam>
+    /// <typeparam name="TNextResult">The type of the next result.</typeparam>
+    /// <param name="errorOr">The error.</param>
+    /// <param name="onValue">The function to execute if the state is a value.</param>
+    /// <returns>The result from calling <paramref name="onValue"/> if state is value; otherwise the original errors.</returns>
+    public static async Task<ErrorOr<TNextResult>> ChainAsync<TResult, TNextResult>(this Task<ErrorOr<TResult>> errorOr, Func<TResult, Task<ErrorOr<TNextResult>>> onValue)

```

Comment:
There could be one more convenience function where there is a `Task<ErrorOr<T>>` given and you apply a non async function to it:

```csharp
public static async Task<ErrorOr<TResult>> Then<T, TResult>(this Task<ErrorOr<T>> errorOrTask, Func<T, ErrorOr<TResult>> mapping)
```

Reply:
Great suggestion. It's a bit odd that it returns `Task<..>`, but I'm sure devs will find it useful.